src/Compilers/Core/Portable/SourceGeneration/IncrementalWrapper.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Wraps an incremental generator in a dummy <see cref="ISourceGenerator"/> interface.
+    /// </summary>
+    /// <remarks>
+    /// Used to unify loading incremental generators with older ISourceGenerator style ones
+    /// </remarks>
+    /// <typeparam name="TIncrementalGenerator">The type of the incrmental generator being wrapped</typeparam>
+    internal class IncrementalToSourceGeneratorWrapper<TIncrementalGenerator> : IncrementalGeneratorWrapper, ISourceGenerator
```

Comment:
What is the value in having this be a generic type vs. just wrapping `IIncrementalGenerator` values?

Reply:
Unfortunately we have several other places in the code (especially the IDE) that assume that there is essentially a 1 to 1 mapping between the instance and it's type. We've never explicitly made that guarantee, but we even have Debug.Asserts that check it here http://sourceroslyn.io/#Microsoft.CodeAnalysis/SourceGeneration/GeneratorDriver.cs,39

If we wrap without including the type we break that assumption, and need a bigger refactoring. I was unsure if we wanted to just go that route, but it seemed like a small enough pain point in the loader to hide it from all the downstream clients.

Reply:
Ah I see. Can we note that in the remarks section then as the reason for it being generic here? Seems fine for a PROTOTYPE comment / future tracking issue (not critical to release).

Reply:
I have no problem with the IDE moving _off_ of any assumptions that way. I've already got a change (https://github.com/dotnet/roslyn/pull/52094/commits/db6c8c45565f83537859707d1f272694710de2da) where we're generally no longer passing Type objects around anyways and just passing strings because having to work with Types is problematic once we start having to do some OOP synchronization.